### ISSUE TITLE
Performance:

### ISSUE THREAD

**Issue by engAhmad** [TeamWertarbyte/crypto-trading-bot#37]
Performance: 
This comes due to the limitation of the bittrex api. It only allows ~60requests per minute. So we have some manual wait time between requests to not get blocked.
Sadly this results in the long total amount. We could see which requests we can omit to speed things up. But for now I don't know which.

Stopping buying new coins:
Maybe there isn't any viable option for buying currently? Adding those shouldn't general break the evaluation of other options.
I'll improve the logging to reflect this behavior. Would you mind opening an issue for that?

_Originally posted by @saschb2b in https://github.com/TeamWertarbyte/crypto-trading-bot/discussions/33#discussioncomment-384610_

**Comment by saschb2b** [comment]
Switching to sockets is a great idea! Will see what their api offers on that connection.
Additionally I noticied we currently iterate over several "dust" balances in our portfolio.
We can (let's see how) detect and skip them to get an additonal boost

**Comment by engAhmad** [comment]
> Additionally I noticied we currently iterate over several "dust" balances in our portfolio.
> We can (let's see how) detect and skip them to get an additonal boost .

What do you mean "dust” ,

the trade remaining ? Or unprofitable coins ?

**Comment by saschb2b** [comment]
Thanks for the invite, joined!

Dust are all assets that are in your portfolio but only worth a few bucks. Like 0.24$ of Digibyte.
Currently the bot sees it and thinks "Well we still got some, so I don't need to buy more" or it tries to sell it due to bad EMA but can't due to minimum trading rate being greater of what you hold.
I had like 4-5 dust assets that "blocked" further transactions. But corrected them manually now.
I think with the current implementation we shouldn't get any more dust coins as we use fill_or_kill for selling. So there is no "partially sold" assets
Tldr: Fixed it manually and we shouldn't care about it anyway I think

**Comment by engAhmad** [comment]
Regrading to Dust Coins issue  , 
i think simply  sell them if they under certain amount specified by the user e.g. under  0.5 USDT  if  this value not reach minimum trading rate so the bot should buy some to reach the trading rate then sell them
